### PR TITLE
Numerical name plugin

### LINKED ISSUE #1319: Flag numerical names on houses
Houses with purely numerical names are normally errors, they should be tagged as `addr:housenumber`

Approximate overpass turbo wizard queries for the selection of elements:

Either
`name~"[0-9]+"` and (`building=house` or `building=detached` etc.

Or
`name~"[0-9]+"` and `building=*` and !(`building=garage` or `shop=*` or `amenity=*` etc.

This could be extended to look for houses with name, which should be `addr:housename`

### PR DESCRIPTION
**Pull request by arrival-spring**
Fixes #1319 

First time contributing and haven't been able to test this.

### CODE REVIEW

**Comment by Famlam** [comment]
About [80% of all buildings](https://taginfo.openstreetmap.org/keys/building#values) is tagged `building=yes` and thus excluded. I'm wondering if a blacklist approach wouldn't be better? (Note: I'm not a maintainer of this repository, it's just my thoughts)

**Comment by frodrigo** [comment]
> About [80% of all buildings](https://taginfo.openstreetmap.org/keys/building#values) is tagged `building=yes` and thus excluded. I'm wondering if a blacklist approach wouldn't be better? (Note: I'm not a maintainer of this repository, it's just my thoughts)

Yes, I'm agree. Why limit to a subset of buildings. And mainly, why exclude building=yes ?

**Comment by arrival-spring** [comment]
My thinking was to try to avoid false positives, but I guess another way to do it would be a blacklist and exclude anything with amenity, leisure, tourism, etc.

**Comment by frodrigo** [comment]
@arrival-spring still interested in finishing this PR ?

**Comment by arrival-spring** [comment]
Yes, sorry, it had dropped off my radar, I'll see if I can finish it this week

**Comment by arrival-spring** [comment]
How's that?
(Still untested)

**Comment by frodrigo** [comment]
Please, can you also review all open comments.

**Comment by arrival-spring** [comment]
I missed a `not`, sorry

I'm not sure which comments you're referring to there

**Comment by frodrigo** [comment]
> I'm not sure which comments you're referring to there

Code comments https://github.com/osm-fr/osmose-backend/pull/1406/files

**Comment by Famlam** [comment]
> > I'm not sure which comments you're referring to there
> 
> Code comments https://github.com/osm-fr/osmose-backend/pull/1406/files

In all honesty I don't see any comments either @frodrigo

**Comment by frodrigo** [comment]
> > > I'm not sure which comments you're referring to there
> > 
> > 
> > Code comments https://github.com/osm-fr/osmose-backend/pull/1406/files
> 
> In all honesty I don't see any comments either @frodrigo

It should be available now.

**Comment by arrival-spring** [comment]
All done

File: plugins/Name_ShouldBeHousenumber.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+class Name_ShouldBeHousenumber(Plugin):
+
+    def init(self, logger):
+        Plugin.init(self, logger)
+        self.errors[804] = self.def_class(item = 5010, level = 1, tags = ['name', 'fix:chair'],
+            title = T_('Name with only numbers'),
+            detail = T_(
+'''This building is tagged with a name which contains only numbers
+and does not appear to have a main feature key.
+ For the majority of buildings this should be tagged as the housenumber.'''),
```
Comment (frodrigo) [inline plugins/Name_ShouldBeHousenumber.py:35]:
```suggestion
For the majority of buildings this should be tagged with `housenumber`.'''),
```

File: plugins/Name_ShouldBeHousenumber.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        self.Numerical = re.compile(r"^[0-9]+$")
+        self.feature_keys = ["amenity", "craft", "emergency", "leisure", "military", "office", "railway", "shop", "tourism"]
+
+    def node(self, data, tags):
+        err = []
+        if "building" in tags:
+            if not any(feature in tags for feature in self.feature_keys):
+                if "name" in tags:
+                    if self.Numerical.match(tags["name"]):
+                        err.append({"class": 804, "text": T_("Concerns tag: `{0}`", '='.join(['name', tags['name']])) })
```
Comment (frodrigo) [inline plugins/Name_ShouldBeHousenumber.py:49]: A fix suggestion could be done in this case. Remove `name`, add `housenumber`.
https://github.com/osm-fr/osmose-backend/blob/master/doc/0-Index.md#fix-suggestion
 Eg: https://github.com/osm-fr/osmose-backend/blob/master/plugins/TagFix_Pharmacy.py#L46

File: plugins/Name_ShouldBeHousenumber.py (hunk 1 above)
Comment (Famlam) [reply plugins/Name_ShouldBeHousenumber.py:35]: I think you mean `addr:housenumber`?

Comment (Famlam) [reply plugins/Name_ShouldBeHousenumber.py:35]: I think you mean `addr:housenumber`?

Comment (frodrigo) [reply plugins/Name_ShouldBeHousenumber.py:35]: yes

**Comment by arrival-spring** [comment]
Makes sense, I've now added the fix suggestion

File: plugins/Name_ShouldBeHousenumber.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+from modules.OsmoseTranslation import T_
+from plugins.Plugin import Plugin
+import regex as re
+
+class Name_ShouldBeHousenumber(Plugin):
+
+    def init(self, logger):
+        Plugin.init(self, logger)
+        self.errors[804] = self.def_class(item = 5010, level = 1, tags = ['name', 'fix:chair'],
```
Comment (Famlam) [inline plugins/Name_ShouldBeHousenumber.py:30]: Just a minor remark: is level 1 (high priority) appropriate? I would personally go for level 2 or 3

Comment (arrival-spring) [reply plugins/Name_ShouldBeHousenumber.py:30]: I'm not sure I know enough about levels to have an opinion

Comment (frodrigo) [reply plugins/Name_ShouldBeHousenumber.py:30]: Yes, not level 1. 2 look good for me.

Comment (arrival-spring) [reply plugins/Name_ShouldBeHousenumber.py]: Done

**Comment by frodrigo** [comment]
Than all four your contribs on this long run.
Merged, but not deployed yet.

**Comment by frodrigo** [comment]
Deployed.

**Comment by frodrigo** [comment]
Hum. Should be here. But nothing.
https://osmose.openstreetmap.fr/en/issues/open?item=5010&class=804&useDevItem=all

**Comment by arrival-spring** [comment]
Hmm. I can't understand why the tests would pass but nothing would be showing up

**Comment by frodrigo** [comment]
I got it class number over all sax plugin should be unique.

```
Exception: class 804 already present as item 2020 
```

I make a fix.

**Comment by frodrigo** [comment]
Now is running.
https://osmose.openstreetmap.fr/en/issues/open?item=5010&class=50101&useDevItem=all
But result are still empty.

**Comment by arrival-spring** [comment]
Well, I have the same comment as before. The tests pass so I don't know why the results are empty.

**Comment by frodrigo** [comment]
Previous fix was partial. Fix it again.

**Comment by Famlam** [comment]
Good catch! I looked over the code 10 times without seeing that

**Comment by frodrigo** [comment]
Now we finally got data!

Welcome in Moldavia
https://osmose.openstreetmap.fr/en/map/#item=5010&class=50101&zoom=17&lat=46.668997&lon=28.764851&level=1%2C2%2C3

**Comment by arrival-spring** [comment]
Great!

**Comment by frodrigo** [comment]
Thank, trap wording added, along an other wording contrib.

**Pull request merged**